src/routes/api/users.ts

```diff
@@ -1,15 +1,66 @@
 import express from "express";
 import authWithAcessMiddleware from "../../middleware/auth/user/authWithAcessMiddleware";
 import authDataOnlyMiddleware from "../../middleware/auth/user/authDataOnlyMiddleware";
+import { User, UserModel } from "../../models/user";
+import bcrypt from "bcrypt";
 
 const userRouter = express.Router();
 
-userRouter.get("/:userId", authDataOnlyMiddleware, (req, res) => {
-  res.send("a specific user");
+userRouter.get("/:userId", authDataOnlyMiddleware, async (req, res) => {
+  try {
+    const userId = req.params.userId;
+    //Find the user in db
+    const userDoc = await UserModel.findById(userId);
+    if (!userDoc) return res.status(404).json({ error: "user doesn't exist" });
+    if (userDoc.private_profile) res.status(200).json({ user: "This user profile is public" });
+
+    res.status(200).json({
+      id: userDoc.id,
+      username: userDoc.username,
+      liked_projects: userDoc.liked_projects,
+    });
+  } catch (err) {
+    console.log(err);
+  }
 });
 
-userRouter.put("/:userId", authWithAcessMiddleware, (req, res) => {
-  res.send("updated users settings");
+userRouter.put("/:userId", authWithAcessMiddleware, async (req, res) => {
+  try {
+    const userId = req.params.userId;
+    const newUserSettings = req.body; //Retreive new settings
+    //Find the user in db
+    const userDoc = await UserModel.findById(userId);
+    if (!userDoc) return res.status(404).json({ error: "user doesn't exist" });
+    // Update user settings
+
+    if (newUserSettings.username) {
+      const userNewUsernameDoc = await UserModel.findOne({ username: newUserSettings.username });
+      if (userNewUsernameDoc) res.status(400).json({ error: "The username is already taken" });
+    }
+
+    let newPassword;
+
+    if (newUserSettings.password) newPassword = <string>await bcrypt.hash(newUserSettings.password, userDoc.password_salt);
```

Comment:
You don't need to cast to string since you're using await and the result will always be a string (otherwise it will throw and there will be no result at all).